### DISCUSSION TITLE
Incomplete specification of eval order in interpolated strings

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/csharplang#5435] [General]
If we take the following code:
```csharp
using System;

var s = $"{GetFoo(1)}{GetFoo(2)}{GetFoo(3)}";

Foo GetFoo(int i)
{
    Console.WriteLine($"Get({i})");
    return new Foo()
    {
        I = i
    };
}

class Foo
{
    public int I;

    public override string ToString()
    {
        Console.WriteLine("ToString()");
        return I.ToString();
    }
}
```

We got in dotnet 5 the following output:
```
Get(1)
Get(2)
Get(3)
ToString()
ToString()
ToString()
```

But with dotnet 6 (interpolated string handler) the behaviour is changed to:
```
Get(1)
ToString()
Get(2)
ToString()
Get(3)
ToString()
```

Could we specify which behaviour is correct? Or if both options are valid it would be nice if we could write it in the spec (like we did with pattern matching).

**Comment by 333fred** [comment]
The latter behavior is correct, and no further updates to the specification are necessary. The specification covers the handler pattern, and demonstrates when calls to the DefaultInterpolatedStringHandler will be made. That is all it can and should do.
